demonstrations/tutorial_qutrits_bernstein_vazirani.py

```diff
@@ -0,0 +1,394 @@
+
+    # Initialize x = [0,1,0]
+    qml.PauliX(wires = 1)
+
+    # We apply our oracle
+    Uf()
+
+    # We measure the last qubit
+    return qml.sample(wires = 3)
+
+@qml.qnode(dev)
+def circuit2():
+    # Circuit used to derive a2
+    # Initialize x = [0,0,1]
+    qml.PauliX(wires = 2)
+
+    # We apply our oracle
+    Uf()
+
+    # We measure the last qubit
+    return qml.sample(wires = 3)
+
+# We run for x = [1,0,0]
+a0 = circuit0()
+
+# We run for x = [0,1,0]
+a1 = circuit1()
+
+# We run for x = [0,0,1]
+a2 = circuit2()
+
+print(f"The value of 'a' is [{a0},{a1},{a2}]")
+
+##############################################################################
+#
+# In this case, with 3 queries (:math:`n=3`), we have discovered the value of :math:`\vec{a}`. Let's run the Bernstein-Vazirani subroutine (using qubits as qubits this time) to check that one call is enough:
+
+
+@qml.qnode(dev)
+def circuit():
+
+    # We initialize to |0001>
+    qml.PauliX(wires = 3)
+
+    # We run the Hadamards
+    for i in range(4):
+        qml.Hadamard(wires = i)
+
+    # We apply our function
+    Uf()
+
+    # We run the Hadamards
+    for i in range(3):
+        qml.Hadamard(wires = i)
+
+    # We measure in the first 3 qubits
+    return qml.sample(wires = range(3))
+
+a = circuit()
+
+print(f"El valor de a es {a}")
+
+##############################################################################
+# Great! Everything works as expected, we have just successfully executed the Bernstein-Vazirani algorithm.
+# It is important to note that, because of how we defined our device, we are only using a single shot to find this value!
+#
+# The generalization to qutrits
+# ------------------------------
+#
+# Now things get more interesting, let's imagine that our basic unit of information is now not the qubit but the qutrit.
+# We can generalize the statement as follows.
+#
+# We are given a function of the form :math:`f(\vec{x}) = \vec{a}\cdot\vec{x} \pmod 3` where :math:`\vec{a}:=(a_0,a_1,...,a_{n-1})` and :math:`\vec{x}:=(x_0,x_1,...,x_{n-1})` are strings of length :math:`n` with :math:`a_i, x_i \in \{0,1,2\}`. How can we minimize the number of calls to the function to discover :math:`\vec{a}`? In this case, the classical procedure to detect the value of :math:`\vec{a}` is the same as in the case of qubits: we will evaluate the output of the inputs :math:`[1,0,0]`, :math:`[0,1,0]` and :math:`[0,0,1]`.
+# To do this we must use qutrit operators. In particular, we will use the :class:`~.pennylane.TShift` which is equivalent to the :class:`~.pennylane.PauliX` gate for qubits. It has the following property:
+#
+# .. math::
+#   \text{TShift}|0\rangle = |1\rangle
+#
+# .. math::
+#   \text{TShift}|1\rangle = |2\rangle
+#
+# .. math::
+#   \text{TShift}|2\rangle = |0\rangle
+#
+# Therefore we can use this gate to initialize each of the states.
+
+
+import pennylane as qml
+from pennylane import numpy as np
+
+dev = qml.device("default.qutrit", wires = 4, shots = 1)

```

Comment:
```suggestion
dev = qml.device("default.qutrit", wires=4, shots=1)

```

Reformat the code blocks here as well